agent/api/container.go

```diff
@@ -370,3 +375,62 @@ func (c *Container) IsEssential() bool {
 func (c *Container) AWSLogAuthExecutionRole() bool {
 	return c.LogsAuthStrategy == awslogsAuthExecutionRole
 }
+
+// SetCreatedAt sets the timestamp for container's creation time
+func (c *Container) SetCreatedAt(createdAt time.Time) {
+	if createdAt.IsZero() {
```

Comment:
When would this case happen? Either way, we do a zero check before setting the field in the container response object. So not sure if this is needed.

Reply:
It'd be extremely rare for this to happen for `Create`. The idea is to prevent this field from being overridden if the client invoking this is passing invalid timestamps. In future, if a JSON unmarshaler for example invokes this with empty timestamp, we're protected.